src/components/Formation/FormationList.tsx

```diff
@@ -0,0 +1,151 @@
+"use client";
+import { useEffect, useState } from "react";
+
+import { Formation, FormationInscription } from "@/models/formation";
+import Tag from "@codegouvfr/react-dsfr/Tag";
+
+import FormationCard from "./FormationCard";
+
+type AudienceCategoryType = {
+    key: string;
+    label: string;
+    type: "audience" | "category";
+    value:
+        | "Design"
+        | "Accessibilité"
+        | "Divers"
+        | "Communication"
+        | "Marketing"
+        | "Tech"
+        | "Produit"
+        | "Nouveaux membres";
+};
+const tags: AudienceCategoryType[] = [
+    {
+        key: "newcomers",
+        label: "Nouveaux arrivants",
+        type: "audience",
+        value: "Nouveaux membres",
+    },
+    {
+        key: "design",
+        label: "Design",
+        type: "category",
+        value: "Design",
+    },
+    {
+        key: "accessibility",
+        label: "Accessibilité",
+        type: "category",
+        value: "Accessibilité",
+    },
+    {
+        key: "divers",
+        label: "Divers",
+        type: "category",
+        value: "Divers",
+    },
+    {
+        key: "communication",
+        label: "Communication",
+        type: "category",
+        value: "Communication",
+    },
+
+    {
+        key: "marketing",
+        label: "Marketing",
+        type: "category",
+        value: "Marketing",
+    },
+    {
+        key: "tech",
+        label: "Tech",
+        type: "category",
+        value: "Tech",
+    },
+    {
+        key: "product",
+        label: "Produit",
+        type: "category",
+        value: "Produit",
+    },
+];
+
+export default function FormationList({
+    inscriptions,
+    formations,
+}: {
+    inscriptions: FormationInscription[];
+    formations: Formation[];
+}) {
+    const [filteredFormations, setFilteredFormations] =
+        useState<Formation[]>(formations);
+    const [selectedFilter, setSelectedFilter] = useState<
+        AudienceCategoryType | undefined
+    >(undefined);
+    const filterFormationWithKey = (tag: AudienceCategoryType) => {
+        if (selectedFilter?.value !== tag.value) {
+            setSelectedFilter(tag);
+        } else {
+            setSelectedFilter(undefined);
+        }
+    };
+    useEffect(() => {
+        if (selectedFilter) {
+            setFilteredFormations(
+                formations.filter((formation) => {
+                    if (selectedFilter.type === "audience") {
+                        return formation.audience?.includes(
+                            selectedFilter.value
+                        );
+                    } else {
+                        return formation.category?.includes(
+                            selectedFilter.value
+                        );
+                    }
+                })
+            );
+        } else {
+            setFilteredFormations(formations);
+        }
+    }, [selectedFilter]);
```

Comment:
A priori ce n'est pas nécessaire de mettre `filteredFormations` dans le state car on peut le déduire directement de `selectedFilter` ?

```suggestion
    const [selectedFilter, setSelectedFilter] = useState<
        AudienceCategoryType | undefined
    >(undefined);
    const filterFormationWithKey = (tag: AudienceCategoryType) => {
        if (selectedFilter?.value !== tag.value) {
            setSelectedFilter(tag);
        } else {
            setSelectedFilter(undefined);
        }
    };
    const filteredFormations: Formation[] = selectedFilter ? formations : 
        formations.filter((formation) => {
                    if (selectedFilter.type === "audience") {
                        return formation.audience?.includes(
                            selectedFilter.value
                        );
                    } else {
                        return formation.category?.includes(
                            selectedFilter.value
                        );
                    }
                })
```